Driverless AI/image-processing-in-driverless-ai/image-processing-in-driverless-ai.md

```diff
@@ -0,0 +1,595 @@
+
+    ![sample-validation-error-images](assets/sample-validation-error-images.png)
+
+     - The above **sample validation errors** display instances when the model predicted wrongly. For example, the top left corner sample shows the model predicting a False (0) case of metastatic cancer when the **True** is (1). 
+
+- Sample Grad-CAM visualization - For our model we observe the following: 
+
+    ![sample-grad-cam-visualization](assets/sample-grad-cam-visualization.png)
+
+    - The **Grad-CAM** visualization samples allow us to see where the model looked when generating a prediction and probability. In the two pair images on the top left corner, we see the images being label as part of the *True* class (1). In this sample, we see that the model observed the middle left part of the image when deciding that this model belongs to the *True* class and that its probability is 0.852. 
+
+**Note**: For time series and Automatic Image Model experiments, you can view detailed insights while an experiment is running or after an experiment is complete by clicking on the **Insights** option.  
+
+Now in the next task, let's compare and contrast each image modeling approach, and let's discuss several scenarios when a given approach will be better. In particular, and as a point of distinction, let's discuss how, between the two approaches, only the **Embeddings Transformer** approach supports a MOJO Scoring Pipeline. 
+
+## Task 6: Final Analysis 
+
+Under what circumstances a particular approach will be better? When answering this question, consider the following:
+
+- When your classification or regression problem is making use of a mixed data type 
+- you can only use the Embeddings Transformer (Image Vectorizer) approach: 
+    - When deciding whether to use it with or without fine-tuning, you can consider what was discussed in tasks 2 and 3. In general, if your dataset is not similar to the ImageNet dataset or we want to improve the results of our model using ImageNet architectures, we can use fine-tuning. 
+        - **Without fine-tuning**: the experiment will usually finish faster but has the lowest performance 
+        - **With fine-tuning**: the experiment will be a bit slower, but should produce better results  
+        - **Automatic Image Model**: the slowest by far, but produces the best results 
+- When your dataset image column is crucial to your regression or classification problem, it is best to use the second approach: Automatic Image Model. Hence, if images are not playing a crucial role in your experiment, you can use the Embeddings Transformer.
+- **Python scoring** and **C++ MOJO Scoring** are supported for the image transformer.
+- Presently, only **Python scoring** is supported for **Automatic Image Model**
+
+With the above in mind, you are ready to generate your Image Models. Note: as of now, Driverless AI supports the following problem types: 
+
+- Embeddings Transformer 
```

Comment:
We don't support `Embeddings Transformer` problem type. We just support the usual DAI's classification (single-label) and regression.